src/lib/getAuditLogger.ts

```diff
@@ -5,6 +5,54 @@ import { isError } from "types/Result"
 import logger from "utils/logger"
 import { UserServiceConfig } from "./config"
 import generateAuditLog from "./generateAuditLog"
+import axios from "axios"
+
+enum HttpStatus {
+  Created = 201
+}
+
+const getApiAuditLogger =
+  (context: GetServerSidePropsContext, config: UserServiceConfig): AuditLogger =>
+  async (action, attributes): PromiseResult<void> => {
+    try {
+      const {
+        timestamp,
+        action: auditAction,
+        username,
+        userIp,
+        requestUri,
+        attributes: auditAttributes
+      } = generateAuditLog(context, action, attributes)
+
+      const userEvent = {
+        eventSource: "User Service",
+        category: "information",
+        eventType: auditAction,
+        timestamp: `${timestamp.toISOString()}`,
+        attributes: {
+          auditLogVersion: "2",
+          eventCode: auditAction.replace(/ /g, "-").toLowerCase(),
```

Comment:
I think we should carefully think out the event codes, and have an enum or something with them in.

We should come up with a list of all of the audit log events that the user-service will generate, and then manually decide on an appropriate event code for each. I'd expect these manual codes to be a bit clearer than ones that are generated by swapping whitespace for hyphens!

Reply:
That makes sense. I'll move the events to an enum.